### PR TITLE
Giada: initialize at 0.15.4

### PR DESCRIPTION
**Pull request by Zer0-**
###### Motivation for this change
Giada is not in nixpkgs, it is one of the more capable sound looping programs on linux right now.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/audio/giada/default.nix [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      longDescription = ''
+        Giada is a free, minimal, hardcore audio tool for DJs, live performers
+        and electronic musicians. How does it work? Just pick up your channel,
+        fill it with samples or MIDI events and start the show by using this
+        tiny piece of software as a loop machine, drum machine, sequencer,
+        live sampler or yet as a plugin/effect host.
+      '';
+      homepage = https://www.giadamusic.com;
+      platforms = platforms.linux;
+      license = lib.licenses.gpl3;
```
Comment (jtojnar) [inline pkgs/applications/audio/giada/default.nix]:
```suggestion
      license = lib.licenses.gpl3Plus;
```

https://github.com/monocasual/giada/blob/251c67c3431b85ef3ac634e7348136cf8843b06d/src/main.cpp#L11-L14

File: pkgs/applications/audio/giada/default.nix [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    };
+
+  src = fetchFromGitHub {
+    owner = "monocasual";
+    repo = "giada";
+    rev = "0e9a7aa5184dc73c6d39e2e5ee6c088c46b28aa6";
```
Comment (jtojnar) [inline pkgs/applications/audio/giada/default.nix]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/audio/giada/default.nix [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    rtmidi
+    xorg.libXpm
+    xorg.libXcursor
+    xorg.libXinerama
+    libjack2
+    alsaLib
+    libpulseaudio
+  ];
+
+  buildPhase = ''
```
Comment (jtojnar) [inline pkgs/applications/audio/giada/default.nix]: You should not be overriding phases unless you know what you are doing. It breaks many features of generic builder like pre/post phase hooks and `configureFlags`.

Instead you should use

```nix
preConfigure = ''
  ${stdenv.shell} ./autogen.sh
'';

configureFlags = [
  "--target=linux"
];
```

The generic builder will take care of running `configure`, `make` and `make install`.

You can also try adding `doCheck = true` to run `make check`.

Also do not forget to add `--enable-system-catch` to `configureFlags`, see https://www.giadamusic.com/documentation-compiling-from-source

File: pkgs/applications/audio/giada/default.nix [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  src = fetchFromGitHub {
+    owner = "monocasual";
+    repo = "giada";
+    rev = "0e9a7aa5184dc73c6d39e2e5ee6c088c46b28aa6";
+    sha256 = "17ggz04gczldhy2lf0v36bgc29dyjpjbxd0f3jj7y4sv2ya5bwv9";
+  };
+
+  buildInputs = [
+    autoconf
```
Comment (jtojnar) [inline pkgs/applications/audio/giada/default.nix]: Programs run during build should go to `nativeBuildInputs`.

File: pkgs/applications/audio/giada/default.nix [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+, jansson
+, rtmidi
+, xorg
+, libjack2
+, alsaLib
+, libpulseaudio
+}:
+
+stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
```
Comment (jtojnar) [inline pkgs/applications/audio/giada/default.nix]: This is no longer necessary, `mkDerivation` does it by default.

**Review by jtojnar — CHANGES_REQUESTED** [review]

**Comment by Zer0-** [comment]
@jtojnar OK, Thank you for the feedback. I've made changes towards good practices.

**Comment by Zer0-** [comment]
"This PR breaks listing of package outputs after merging. " what does this mean?

**Comment by aanderse** [comment]
@GrahamcOfBorg eval

File: pkgs/applications/audio/giada/default.nix [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      longDescription = ''
+        Giada is a free, minimal, hardcore audio tool for DJs, live performers
+        and electronic musicians. How does it work? Just pick up your channel,
+        fill it with samples or MIDI events and start the show by using this
+        tiny piece of software as a loop machine, drum machine, sequencer,
+        live sampler or yet as a plugin/effect host.
+      '';
+      homepage = https://www.giadamusic.com;
+      platforms = platforms.linux;
+      license = lib.licenses.gpl3Plus;
```
Comment (aanderse) [inline pkgs/applications/audio/giada/default.nix]: remove `lib.`.

**Review by aanderse — CHANGES_REQUESTED** [review]
@Zer0- looks like an evaluation error. Please correct.

**Comment by prusnak** [comment]
Giada has been packaged since cb0b0e93a380d85d901b056b5df3570841899678

**Pull request closed without merging**